### PR TITLE
Implement `NullableTrait`

### PR DESCRIPTION
**Pull request by roby2014**
# Description

- Implement `NullableTrait` trait
- Tests
- Sample 

## Checklist

- [X] Self-review changes.
- [X] Evaluate impact on the documentation.
- [X] Ensure test coverage.
- [X] Write new samples.

### CODE REVIEW

**Review by RiscadoA — APPROVED** [review]
Looks good to me! :) Nice work, thanks!

**Comment by RiscadoA** [comment]
Also, it would be cool to integrate this with the data inspector later on, I will open an issue for it :relieved:

File: core/include/cubos/core/reflection/traits/nullable.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{
+    /// @brief Stores the null representation of a type'.
+    /// @see See @ref examples-core-reflection-traits-nullable for an example of using this trait.
+    /// @ingroup core-reflection
+    class NullableTrait
+    {
+    public:
+        /// @brief Function pointer to check if object is null.
+        /// @param instance Instance.
+        /// @return Whether the object is null.
```
[the remark this answers is not in the document]
Comment (roby2014) [reply core/include/cubos/core/reflection/traits/nullable.hpp:19]: Wait why

[the remark this answers is not in the document]
Comment (RiscadoA) [reply core/include/cubos/core/reflection/traits/nullable.hpp]: It isn't valid, we cant document function pointer alias arguments ig

[the remark this answers is not in the document]
Comment (roby2014) [reply core/include/cubos/core/reflection/traits/nullable.hpp]: I added a separate commit to remove every function pointer doc in the traits headers

**Pull request merged**